### PR TITLE
[WIP] Gatsby migration fixes

### PR DESCRIPTION
**Pull request by pavelgrinchenko**
Fixes related to issues that we got after migrating to Gatsby.js in [this PR](https://github.com/iterative/dvc.org/pull/1025)

- [x]  404 code is not returned on 404
- [x] if I open https://dvc.org/doc/tutorials/versioning#automating-capturing Automatic Capturing is not highlighted on the right sidebar (and it’s not even active)
- [x]  go to the last anchor https://dvc.org/doc/tutorials/versioning#whats-next and then click Interactive Tutorial - behavior between previous version and the current one should be the same (no long scrolling animation)
- [x] styles are broken (different) in the new version - go carefully through all of them - make sure that docs are identical, inline code blocks, line spacing, list items spacing, notes, notes, cases like titles with links, etc, etc
- [x] in a Phablet mode Edit on Github button creates a second line (it might be related to styles issue - we already have two lines in the new version, compared to the previous one)
- [x] SEO problems - run Lighthouse new and old version
- [x] image subtitles are broken - check this page https://dvc.org/doc/tutorials/versioning , (Who doesn’t love ASCII directory art?)
- [x] make sure that GA runs the same way as before, with all the events
- [x] check Sentry - we are getting some new errors, hard to tell what page they come from, from what action

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,19 +1,47 @@
...
-app.use(notFoundMiddleware)
+app.use((req, res) => {
+  serveHandler(req, res, {
+    public: 'public',
+    cleanUrls: true,
+    trailingSlash: false,
+    directoryListing: false,
+    headers: [
+      {
+        source: '**/*.@(jpg|jpeg|gif|png)',
```
Comment (shcheklein) [inline server.js]: @fabiosantoscode could you please review this?

File: src/utils/resources.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+export const loadResource = url => {
+  if (!resourcesCache[url]) {
+    resourcesCache[url] = new Promise((resolve, reject) => {
+      const howToHandle = Object.keys(resourceNodeCreators).find(regExp =>
+        new RegExp(regExp).test(url)
+      )
+
+      console.log(url, howToHandle)
```
Comment (shcheklein) [inline src/utils/resources.js]: is it a debug?

Comment (pavelgrinchenko) [reply src/utils/resources.js]: Yes. Removed it.

**Comment by pavelgrinchenko** [comment]
> make sure that GA runs the same way as before, with all the events

I've checked few events and they was sent. So all seems good.

**Comment by pavelgrinchenko** [comment]
> check Sentry - we are getting some new errors, hard to tell what page they come from, from what action

Most critical issues were fixed. The rest needs to be watched.

File: server.js (hunk 1 above)
Comment (shcheklein) [reply server.js]: @fabiosantoscode ??

**Pull request merged**